### ISSUE TITLE
Unexpected token error - keys in list function

### ISSUE THREAD

**Issue by mablo** [phpmd/phpmd#512]
In php 7.1 was added the ability to add keys in the list function (https://wiki.php.net/rfc/list_keys).

Example code:
```php
<?php
list (
    'foo' => $foo,
    'bar' => $bar,
) = [
    'foo' => 'FOO',
    'bar' => 'BAR',
    'baz' => 'BAZ'
];
var_dump($foo, $bar);
```

Error:
```Unexpected token: 'foo', line: 4, col: 5, file: test.php.```

**Comment by Zakerius** [comment]
I have something similar in foreach loop:

```
foreach ($foo as [
    'a' => $a,
    'b => $b
]) {
    ...
}
```
gives me:

> Unexpected token: [

should it be separate issue?

**Comment by anyt** [comment]
Same issue with `forach`, are there any plans to fix it in nearest feature?

**Comment by HEKET313** [comment]
Any info about this bug? When can we expect fix? @ravage84 @manuelpichler

**Comment by MichalKatuscak** [comment]
I have the same issue. Will be fixed?

**Comment by ThomasLabstep** [comment]
Any ETA?

**Comment by tvbeek** [comment]
This is probably an issue that also need a change in [pdepend](https://github.com/pdepend/pdepend/), feel welcome to create a PR to fix this.

**Comment by kylekatarnls** [comment]
This will be fixed in https://github.com/pdepend/pdepend/pull/383
As you can see here:
https://github.com/pdepend/pdepend/pull/383/files#diff-eea33dc8e05dbef887669deda40a18e5

**Comment by kylekatarnls** [comment]
I forgot to close this issue. It has been fixed in PDepend 2.6.0.
